consensus/obcpbft/pbft-core.go

```diff
@@ -1266,3 +1280,28 @@ func (instance *pbftCore) stopTimer() {
 	instance.timerActive = false
 	instance.newViewTimer.Stop()
 }
+
+type requestSlice []*Request
+
+func (a requestSlice) Len() int {
+	return len(a)
+}
+func (a requestSlice) Swap(i, j int) {
+	a[i], a[j] = a[j], a[i]
+}
+func (a requestSlice) Less(i, j int) bool {
+	if a[i].Timestamp == nil {
+		// a[i] has no timestamp, handle it later, TODO, eventually this should be an error
+		return false
+	}
+
+	if a[j].Timestamp == nil {
+		// a[j] has no timestamp, handle it later, TODO, eventually this should be an error
+		return true
+	}
+
+	iTime := time.Unix(a[i].Timestamp.Seconds, int64(a[i].Timestamp.Nanos))
```

Comment:
At the very least perhaps some utility functions for converting to and from the protobuf Timestamp structure.